### PR TITLE
Fix point vs line subtype drawing bug 

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/9841

In prod right now, the way we decide if the drawing tool should be line or point is we have an array of the component types that are represented by lines and then compare the type selected to that array. The problem is that Highway is usually line except for the Access Ramp subtype, which is a point. However when you try to draw the access ramp, it still forces lines. 

So instead of an array, this PR makes an object with the keys being component type + component subtype, and when a type or subtype is selected, we lookup on the object. 

```
	"guardrail":true,
	"highwayaccessramp":false,
	"highwayaddedcapacity/lanes":true,
	"highwaycollectordistributor":true,
	"highwayflyover":true,
	"highwayintersectiongradeseparation":true,
	"highwaymanagedlane":true,
	"highwaytollroad":true,
	"landscaping":true,
	"placemaking":true,
	"refugeislandbike":false,
	"refugeislandped":false,
	"refugeislandbike/ped":false,
```

I investigated just accessing the attribute directly and not having to compare to a list, but the way we it set up now is selecting from a string list of options in the Autocomplete dropdown and not having the whole object for the type accessible.  I'm hesitant to work on a big refactor here since we will be updating our mapping soon anyway (https://github.com/cityofaustin/atd-data-tech/issues/9675) and I dont know what changes will be needed there. 

## Testing
**URL to test:** 
https://deploy-preview-732--atd-moped-main.netlify.app/moped/projects/1862?tab=map

**Steps to test:**
Add a new component to the map. 
Select Highway as your type. 
Select some subtypes, and see that they are lines except access ramp. 

Try around other types and subtypes toggling back and forth. 

---
#### Ship list
- [x] Code reviewed 
- [ ] Product manager approved
- [x] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js:726]: fixing a warning/error in the console about deprecated props

File: moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js:261]: I removed `e.target.value` because our `<li>` does not have a value attribute and e.target.value is `0`. If there wasn't a newValue, we would be doing `0.toLowerCase()` which results in an error.

File: moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js
Comment (frankhereford) [inline moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js:261]: Thank you for explaining to me the reasoning behind doing this comparison using the combined names as a key to lookup instead of pushing this back to the database to resolve.

**Review by frankhereford — APPROVED** [review]
Thank you for finding and fixing this -- and for squashing a console error too! 🚀 

I've looked at the code and tested this locally and it works just as you described.

File: moped-editor/src/utils/projectComponentHelpers.js
Comment (mddilley) [inline moped-editor/src/utils/projectComponentHelpers.js:55]: This feels like a good place to create a helper that describes how this translation works to help ID components/subtypes, but I think that you are right to hold off on too much work here before potential refactoring. 👍

**Review by mddilley — APPROVED** [review]
Works as described, and TIL how this part of the app works. 🙏

**Comment by chiaberry** [comment]
After merging it's parent branch to main, this PR can now be tested on netlify: https://deploy-preview-732--atd-moped-main.netlify.app/moped/projects/1862?tab=map @patrickm02L

**Review by patrickm02L — APPROVED** [review]
It's good to go! 🤘

https://user-images.githubusercontent.com/5566727/183949676-d9988a4a-90fa-4255-877f-7e47ee3e726e.mp4

**Pull request merged**
